File: src/linkerz_crawler_core/src/main/java/org/linkerz/crawler/core/queue/Queue.java
package org.linkerz.crawler.core.queue;

import org.linkerz.crawler.core.job.Job;

import java.io.Serializable;

/**
 * The Class Queue.
 *
 * @author Nguyen Duc Dung
 * @since 7/3/12, 3:16 AM
 */
public interface Queue<J extends Job> extends Serializable {

    /**
     * Add a job to the queue.
     * @param job
     */
    void add(J job);

    /**
     * Run the queue.
     */
    void run();

}

File: src/linkerz_crawler_core/src/main/scala/org/linkerz/crawler/core/queue/Queue.scala
package org.linkerz.crawler.core.queue

import org.linkerz.crawler.core.job.Job

/**
 * The Class Queue.
 *
 * @author Nguyen Duc Dung
 * @since 7/2/12, 11:37 PM
 *
 */

trait Queue {

  /**
   * Add a job to the queue.
   * @param job
   */
  def add(job: Job)


  /**
   * Run the queue.
   */
  def run()

}
